app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+class LikesController < ApplicationController
+  def create
+    @likeable = params[:likeable_type].constantize.find(params[:likeable_id])
+    @like = @likeable.likes.create!(user_id: current_user.id)
+    respond_to_format
+  end
+
+  def destroy
+    @like = current_user.likes.find(params[:id]).destroy
+    @likeable = @like.likeable_type.constantize.find(@like.likeable_id)
+    respond_to_format
```

Comment:
Let's not define a variable value depending on a record just destroyed. It won't be an issue here but can be depending on the situation.